### PR TITLE
add .gitignore file support

### PR DESCRIPTION
**Pull request by RAFAQAT11**
previously graph init does not create .gitignore file
With this pull request, graph init will be able to create .gitignore file
changes are made in src/commands/init.js file, gitignoreFileContent string & createGitignore function are written.
createGitignore function is placed in two functions (initSubgraphFromExample & initSubgraphFromContract) at appropriate position after the initRepository function.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@graphprotocol/graph-cli",
-  "version": "0.30.0-alpha.1",
+  "version": "0.30.0",
```
Comment (evaporei) [inline package.json:3]: We don't want to change this yet.

File: src/commands/init.js [hunk 2]
```diff
@@ -580,6 +617,23 @@ const initRepository = async (toolbox, directory) =>
     },
   )
 
+const createGitignore = async (toolbox, directory) =>
+  await withSpinner(
+    `Create .gitignore file`,
+    `Failed to create .gitignore file`,
+    `Warnings while creating .gitignore file`,
+    async (spinner) => {
+      // Remove .gitignore file if exist else create .gitignore
```
Comment (evaporei) [inline src/commands/init.js:626]:
```suggestion
      // Remove .gitignore file if it exists and create a new one
```

**Comment by rdotterer09-crypt** [comment]
I want the name Dal-Tiff on this contract

**Comment by raulrpearson** [comment]
I was about to open a PR to add this feature, is this PR moving forward? Is there something I can do to help?

**Pull request closed without merging**